documentation/src/main/asciidoc/chinese/chapter-8-值的映射.asciidoc

```diff
@@ -0,0 +1,268 @@
+[[mapping-enum-types]]
+== 值的映射
+
+=== 把enum映射到enum
+
+MapStruct支持生成的方法实现中，把Java里的enum类型映射成另一个enum类型。
+
+默认情况下，source enum的每个常量会被映射为target enum中名字相同的对应常量。如果有需要，您可以通过 `@ValueMapping` 来把source enum中的常量映射为其他名字。source enum中的多个常量可以被映射为target enum中的同一个常量。
+
+例子如下：
+
+.Enum mapping method
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+@Mapper
+public interface OrderMapper {
+
+    OrderMapper INSTANCE = Mappers.getMapper( OrderMapper.class );
+
+    @ValueMappings({
+        @ValueMapping(target = "SPECIAL", source = "EXTRA"),
+        @ValueMapping(target = "DEFAULT", source = "STANDARD"),
+        @ValueMapping(target = "DEFAULT", source = "NORMAL")
+    })
+    ExternalOrderType orderTypeToExternalOrderType(OrderType orderType);
+}
+----
+====
+
+.Enum mapping method result
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+// GENERATED CODE
+public class OrderMapperImpl implements OrderMapper {
+
+    @Override
+    public ExternalOrderType orderTypeToExternalOrderType(OrderType orderType) {
+        if ( orderType == null ) {
+            return null;
+        }
+
+        ExternalOrderType externalOrderType_;
+
+        switch ( orderType ) {
+            case EXTRA: externalOrderType_ = ExternalOrderType.SPECIAL;
+            break;
+            case STANDARD: externalOrderType_ = ExternalOrderType.DEFAULT;
+            break;
+            case NORMAL: externalOrderType_ = ExternalOrderType.DEFAULT;
+            break;
+            case RETAIL: externalOrderType_ = ExternalOrderType.RETAIL;
+            break;
+            case B2B: externalOrderType_ = ExternalOrderType.B2B;
+            break;
+            default: throw new IllegalArgumentException( "Unexpected enum constant: " + orderType );
+        }
+
+        return externalOrderType_;
+    }
+}
+----
+====
+
+当发生以下情况时，MapStruct会报错：
+1. 如果source enum类型中的一个常量在target enum类型中没有名字相同的对应常量，也没有通过 `@ValueMapping` 映射为另一个常量，就会抛出一个错误。这点确保了所有常量转换都是安全的、可预期的。
+2. 如果因为某些原因出现了一个未识别到的source值，同样会抛出IllegalStateException异常。
+
+MapStruct也可以把所有剩余的（未指明的）映射都映射成默认值。这在一组值的映射中只能用一次，而且只能应用于source。有两种选择：`<ANY_REMAINING>` 和 `<ANY_UNMAPPED>`，这两个选项不能同时使用。
+
+在source标签上使用 `<ANY_REMAINING>` 时，MapStruct首先仍会把source enum和target enum中名字相同的常量做映射，之后source enum中剩余的的常量会被映射为该 `@ValueMapping` 中指定的target常量。
+
+当使用 `<ANY_UNMAPPED>` 时，MapStruct *不会* 尝试进行基于名称的映射，而是直接把剩下的常量映射为该 `@ValueMapping` 中指定的target常量。
+
+MapStruct可以通过指定 `<NULL>` 关键字来处理source为 `null` 或者target为 `null` 的情况。
+
```

Comment:
Would you mind also adding the THROW_EXCEPTION options introduced here:

https://github.com/mapstruct/mapstruct/commit/62c7ce1cdf0ccc5528dca84decaf07cbecadaed4#diff-aaca9b52e434c4fb6940e4405327fe73da5f3438442109c4c8c6f858c0d6ddee